application/controllers/ServicesController.php

```diff
@@ -198,6 +201,91 @@ public function searchEditorAction()
         $this->setTitle(t('Adjust Filter'));
     }
 
+    public function gridAction()
+    {
+        $db = $this->getDb();
+        $this->setTitle(t('Service Grid'));
+
+        $query = Service::on($db)->with([
+            'state',
+            'host',
+            'host.state'
+        ]);
+
+        $gridcols = [
+            'host_name' => 'host.display_name',
+            'host_display_name' => 'host.name',
+            'service_name' => 'service.name',
+            'service_display_name' => 'service.display_name',
+            'service_handled' => 'service.state.is_handled',
+            'service_output' => 'service.state.output',
+            'service_state' => 'service.state.soft_state'
+        ];
+
+        $filter = (bool) $this->params->shift('problems', false) ?
+            Filter::equal('service.state.is_problem', 'y') : null;
+
+        $problemToggle = (new CompatForm())->addAttributes(['class' => 'inline']);
+
+        $problemToggle->addElement('checkbox', 'problems', [
+            'class'     => 'autosubmit',
+            'id'        => 'problems',
+            'label'     => $this->translate('Problems Only'),
+            'value'     => $filter !== null
+        ]);
+
+        $this->getRequest()->protectId('problems');
+        $problemToggle->on(Form::ON_SUCCESS, function (Form $form) {
+            if (! $form->getElement('problems')->isChecked()) {
+                $this->redirectNow(Url::fromPath('icingadb/services/grid'));
+            } else {
+                $this->redirectNow(Url::fromPath('icingadb/services/grid?problems'));
+            }
+        });
+
+        $problemToggle->handleRequest(ServerRequest::fromGlobals());
+
+        $this->addControl($problemToggle);
+
+        $sortControl = $this->createSortControl($query, [
+            'host.name'     => $this->translate('Host Name'),
+            'service.name'          => $this->translate('Service Name')
+        ]);
+
+        $searchBar = $this->createSearchBar($query, [
+            $sortControl->getSortParam(),
```

Comment:
You should also add all the params specific to this action. Otherwise they show up in the search bar after using the joystick, flipping the grid or using the toggle slider.